RecoTauTag/RecoTau/interface/PFRecoTauClusterVariables.h

```diff
@@ -1,163 +1,232 @@
+    /// which are associated to signal
+    float tau_pt_weighted_dphi_strip(const reco::PFTau& tau, int dm) const {
+      return pt_weighted_dx(tau, dm==10 ? 2 : 1, 2, dm);
+    }
+    float tau_pt_weighted_dphi_strip(const pat::Tau& tau, int dm) const {
+      return pt_weighted_dx(tau, dm==10 ? 2 : 1, 2, dm);
+    }  
+    /// return sum of pt weighted values of dr relative to tau candidate for all pf photon candidates,
+    /// which are inside an isolation conde but not associated to signal
+    float tau_pt_weighted_dr_iso(const reco::PFTau& tau, int dm) const {
+      return pt_weighted_dx(tau, 2, 0, dm);
+    }
+    float tau_pt_weighted_dr_iso(const pat::Tau& tau, int dm) const {
+      return pt_weighted_dx(tau, 2, 0, dm);
+    }
+    /// return total number of pf photon candidates with pT>500 MeV, which are associated to signal
+    unsigned int tau_n_photons_total(const reco::PFTau& tau) const {
+      unsigned int n_photons = 0;
+      for (auto& cand : tau.signalPFGammaCands()) {
+        if ((float)cand->pt() > 0.5)
+          ++n_photons;
+      }
+      for (auto& cand : tau.isolationPFGammaCands()) {
+        if ((float)cand->pt() > 0.5)
+          ++n_photons;
+      }
+      return n_photons;
+    }
+    unsigned int tau_n_photons_total(const pat::Tau& tau) const {
+      unsigned int n_photons = 0;
+      for (auto& cand : tau.signalGammaCands()) {
+        if (cand->pt() > 0.5) 
+          ++n_photons;
+      }
+      for (auto& cand : tau.isolationGammaCands()) {  
+        if (cand->pt() > 0.5)
+	  ++n_photons;
+      }
+      return n_photons;
     }
-  }
-  
-  float tau_pt_weighted_dr_iso(const reco::PFTau& tau, int dm) {
-    return pt_weighted_dx(tau, 2, 0, dm);
-  }
   
-  unsigned int tau_n_photons_total(const reco::PFTau& tau) {
-    unsigned int n_photons = 0;
-    for (auto& cand : tau.signalPFGammaCands()) {
-      if (cand->pt() > 0.5)
-        ++n_photons;
-    }
-    for (auto& cand : tau.isolationPFGammaCands()) {
-      if (cand->pt() > 0.5)
-        ++n_photons;
-    }
-    return n_photons;
-  }
-
-}
+  private:
+    /// return pf photon candidates that are associated to signal
+    const std::vector<reco::PFCandidatePtr>& getPFGammas(const reco::PFTau& tau, bool signal = true) const {
+      if (signal){
+        return tau.signalPFGammaCands();
+      }
+      return tau.isolationPFGammaCands();
+    }
+    reco::CandidatePtrVector getGammas(const pat::Tau& tau, bool signal = true) const {
+      if(signal){
+        return tau.signalGammaCands();
+      }
+      return tau.isolationGammaCands();
+    }
+    /// decide if photon candidate is inside the cone to be associated to the tau signal
+    bool isInside(float photon_pt, float deta, float dphi) const {
+      double stripEtaAssociationDistance_0p95_p0 = 0.197077;
+      double stripEtaAssociationDistance_0p95_p1 = 0.658701;
+      double stripPhiAssociationDistance_0p95_p0 = 0.352476;
+      double stripPhiAssociationDistance_0p95_p1 = 0.707716;
```

Comment:
Hi @roger-wolf - could you take care of adding these consts to this PR? otherwise, its would be useful to write a comment in the code about from where the numbers were derived.